### PR TITLE
Local config over grpc

### PR DESCRIPTION
**Pull request by timfallmk**
Full fix for #145 and related issues. Signaling for the local `syncthing` process should not be handled by the local `grpc` server, similar to the way it works remotely.

### CODE REVIEW

**Comment by timfallmk** [comment]
@grampelberg This is ready for review. Local debounce is working (and set to 10 seconds for now), but it seems we get a similar problem on the remote end, where cycling it too fast causes it to lose folders, which aren't recognized when it comes back up.

We may be able to solve it with a similar setup, but I'll let you take a look into that.

File: pkg/input/debounce.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package input
```
Comment (grampelberg) [inline pkg/input/debounce.go]: Delete?

File: pkg/syncthing/server.go [hunk 2]
```diff
@@ -82,16 +82,21 @@ func (s *Server) Refresh() error {
...
+// Update takes the current config, sets the server's config to that.
 func (s *Server) Update() error {
 	if _, err := s.client.NewRequest().
 		SetBody(s.Config).
 		Post("system/config"); err != nil {
 		return err
 	}
 
-	return s.Restart()
+	if _, err := s.client.NewRequest().
```
Comment (grampelberg) [inline pkg/syncthing/server.go]: I don't think you need this?

File: pkg/ksync/server/grpc.go [hunk 3]
```diff
@@ -35,7 +38,20 @@ func Listen(list *ksync.SpecList, bind string, port int) error {
 		"port": port,
 	}).Info("listening")
 
-	server := &ksyncServer{SpecList: list}
+	syncthingServer, err := syncthing.NewServer(fmt.Sprintf("localhost:%d", viper.GetInt("syncthing-port")), viper.GetString("apikey"))
```
Comment (grampelberg) [inline pkg/ksync/server/grpc.go]: Nice!

File: pkg/syncthing/server.go (hunk 2 above)
Comment (timfallmk) [reply pkg/syncthing/server.go]: If the path is different, I think I need a new client no? I guess I could just reuse the current one.

Comment (timfallmk) [reply pkg/syncthing/server.go]: I'll keep it for easy of reading for now I guess.

**Pull request merged**